### ISSUE TITLE
Export PDF des processus

### ISSUE THREAD

**Issue by Qlaize** [lairdubois/lairdubois#18]
Bonjour Boris,
Il me semble qu'il pourrait être pas mal de pouvoir exporter le schéma des processus (au moins pour ceux que l'on créé...) dans un format imprimable PDF par exemple. Qu'en penses-tu ?

**Comment by bbeaulant** [comment]
Salut Quentin,
Oui, ça serait une bonne chose ! Je n'ai malheureusement pas trouvé la solution simple pour l'instant de le faire. Il faudrait peut-être repenser le système de rendu pour permettre l'intégration de cette fonctionnalité plus facilement.

**Comment by hedjour** [comment]
Le faire en une sorte d'imprime écran ou permettre un export au format svg ?

**Comment by bbeaulant** [comment]
@hedjour, oui, c'est ce que j'ai essayé. Mais actuellement l'affichage des diagramme est un mix entre des traits SVG et des div HTML. Et pour l'instant la CSS pour le print ne les traite pas bien.
Il serait bon d'éviter d'avoir à redévelopper un moteur de rendu des diagrammes juste pour l'impression. 
En l'état, il manque un petit quelque chose pour que ça fonctionne.

**Comment by hedjour** [comment]
Du coup petite question connais tu Tickz ? ce peut être une forme d'export ?

**Comment by Qlaize** [comment]
Salut Boris, ok je vois que ça n'est pas si évident. Vu de loin je pensais que ce serait plus simple :-)
à+

**Comment by bbeaulant** [comment]
Non, mais ça voudrait donc bien dire refaire un rendu spéciale pour l'impression. 
Il me semble bien que le rendu écran serve directement au rendu impression, non ?
Sinon, à chaque modif, il faut changer le code partout.

**Comment by bbeaulant** [comment]
@Qlaize, j'ai une solution simple pour sortir les graph en PNG. Avec PhantomJS qui capture l'écran.
Et c'est ce qui est utilisé pour faire la miniature des processus.
Sauf que comme un diagramme à une taille potentiellement infinie, il faut bien donné une limite à l'image. Ce qui ne serait ps un soucis si l'export se faisait en vectoriel ...

**Comment by hedjour** [comment]
> 
> 
> Non, mais ça voudrait donc bien dire refaire un rendu spéciale pour l'impression.
> Il me semble bien que le rendu écran serve directement au rendu impression, non ?
> Sinon, à chaque modif, il faut changer le code partout.
> Et surtout ça s'affiche pareil que ça s'imprime ou s'exporte.

C'est pas faux du coup la meilleure solution semble tout faire en svg dès le départ ?
Pour qu'elle raison mixte tu du svg et de Html ?

**Comment by bbeaulant** [comment]
> Pour qu'elle raison mixte tu du svg et de Html ?

Principalement le rendu des textes à taille variable. C'est tout fait en HTML et à calculer en SVG.

**Comment by bbeaulant** [comment]
Je ne dis pas que c'est impossible, c'est qu'il faut prendre le temps de s'y pencher en ayant en tête le truc dans son ensemble. Et honnêtement, j'ai développé ce module pour avoir des graphes dynamiques, pas pour que ça retombe sur du papier. Alors à cette heure, même si j'entends la demande plusieurs fois formuler, c'est comme faire de ce module un logiciel de dessins de diagrammes ... ça il en existe déjà d'autres ... Qui sortiront tout ça sur papier sans problème ;)

**Comment by hedjour** [comment]
C'est vrai je comprends tout à fait ta dynamique.
Je comprends aussi les gars qui comme moi apprécie énormément la plateforme que tu as lancé et toute la communauté qui est derrière, et du coup ont pas trop envie de changer à nouveau d'outils et qui n'ont pas obligatoirement un accès internet dans leur atelier.

Et du coup ne serait il pas possible de faire un export en svg avec une phase "nettoyage" du code html ?
